clients/drcachesim/tests/memref_gen.h

```diff
@@ -137,6 +146,34 @@ gen_exit(memref_tid_t tid)
     return memref;
 }
 
+inline std::vector<memref_t>
+get_memrefs_from_ir(instrlist_t *ilist, std::vector<memref_instr_t> &memref_instr_vec,
+                    const addr_t base_addr = BASE_ADDR)
+{
+    static const int MAX_DECODE_SIZE = 1024;
+    byte decode_buf[MAX_DECODE_SIZE];
+    byte *pc =
+        instrlist_encode_to_copy(GLOBAL_DCONTEXT, ilist, decode_buf,
+                                 reinterpret_cast<app_pc>(base_addr), nullptr, true);
+    assert(pc != nullptr);
```

Comment:
From offline discussion: check for buffer overflow

Reply:
Done.